src/views/Profile/AdminTable/AdminDatePicker.tsx

```diff
@@ -32,18 +35,100 @@ interface IMaterialsDate {
   setChanges: (...params: any[]) => void;
 }
 
+interface ISelectedDay {
+  position: string;
+  bgColor: string;
+  value: string;
+}
+
+const datePickerReducer = (
+  state: {
+    firstSelectedDay: ISelectedDay;
+    secondSelectedDay: ISelectedDay;
+  },
+  action: { type: string; payload: ISelectedDay },
+): {
+  firstSelectedDay: ISelectedDay;
+  secondSelectedDay: ISelectedDay;
+} => {
+  switch (action.type) {
+    case 'SET_SELECTED_DAY':
+      if (action.payload.position === 'start') {
+        return {
+          ...state,
+          firstSelectedDay: {
+            ...state.firstSelectedDay,
+            bgColor: action.payload.bgColor,
+            value: action.payload.value,
+          },
+        };
+      }
+      if (action.payload.position === 'end') {
+        return {
+          ...state,
+          secondSelectedDay: {
+            ...state.secondSelectedDay,
+            bgColor: action.payload.bgColor,
+            value: action.payload.value,
+          },
+        };
+      }
+      break;
+    default:
+      return state;
+  }
+  return state;
+};
+
 export const AdminDatePicker: React.FC<IMaterialsDate> = ({
   start,
   end,
   setChanges,
 }) => {
+  const [datePickerState, dispatch] = useReducer(datePickerReducer, {
+    firstSelectedDay: { position: 'start', bgColor: 'transparent', value: '' },
+    secondSelectedDay: { position: 'end', bgColor: 'transparent', value: '' },
+  });
+
   const classes = useStyles();
+  const datepickerClassesStart = useStylesForDatePicker({
+    bgColorForDatePicker: datePickerState.firstSelectedDay.bgColor,
+  });
+  const datepickerClassesEnd = useStylesForDatePicker({
+    bgColorForDatePicker: datePickerState.secondSelectedDay.bgColor,
+  });
+
   const { t } = useTranslation();
-  const handleDatePicker = (option: string, value) => {
-    setChanges({
-      option,
-      date: requestDate(value),
-    });
+  const today = new Date();
+
+  const handleDatePicker = (option: string, value: Date | null) => {
+    if (value) {
+      const changedValue = value.setHours(23, 59, 59, 0).toString();
+      if (
```

Comment:
I'm a bit lost here. Could you please add some short description why do we check what we check here?

Reply:
Okay. I'll try. Value can be Date, or null. In this line:
                onChange={(value) => {
                  handleDatePicker('start', value);
                }}
                We had an error
                Argument of type 'MaterialUiPickersDate' is not assignable to parameter of type 'Date'.
  Type 'null' is not assignable to type 'Date'.

Reply:
well, this explains `if (value)` part but not `option ==='end' && changedValue != ....`

also I'm not in the context why `setChanges` gets just a single value. does the parent element also have "start + end dates"? does it make sense to rely just on the state in parent instead of having 2 different sources of the truth?